### PR TITLE
A fine-tune example on caltech101 from pre-trained vgg16 net

### PR DESCRIPTION
**Pull request by taoari**
Add a fine-tune example on caltech101 from pre-trained vgg16 net
-----------

`fine-tune/train_imagenet.py` additionally accepts three arguments --dataset='imagenet', --finetune-from=None, --finetune-lr-scale=10. 

If args.dataset == 'imagenet', it is in training mode, otherwise it is in fine-tuning mode. And the last fully connected layers are automatically renamed to `'fc_<dataset>'` if args.dataset != 'imagenet'. The finetune layers are also automatically scaled by args.finetune_lr_scale. 

A simple examples is:

```
python train_imagenet.py --data-dir=caltech101 --dataset=caltech101 --num-classes=102 --num-examples=7315 --model-prefix=finetune_caltech101_vgg16 --finetune-from=vgg16/vgg16-0001 --network=vgg16 --batch-size=20 --lr=0.0005 --gpus=0
```

### CODE REVIEW

File: python/mxnet/model.py [hunk 1]
```diff
@@ -456,6 +456,25 @@ def _check_arguments(self):
         assert(self.symbol is not None)
         self.argument_checked = True
 
+        # skip invalid arg_arams and aux_params
```
Comment (piiswrong) [inline python/mxnet/model.py]: Use mx.init.Load() initializer for this. No need to modify model.py

Comment (taoari) [inline python/mxnet/model.py]: It is OK, but it is not convenient to automatically separate out the fully connected fine-tune layers, and make their learning rate 10 times larger.

Current API for fine-tuning requires manually hacking on the codes, while this PR makes the whole process automatic.
